services/agents/activities/agent-time-usage.ts

```diff
@@ -0,0 +1,102 @@
+import { agentActivities, agents, db, subscriptions, teams } from "@/drizzle";
+import { and, eq, gte, lt, sql } from "drizzle-orm";
+import { getMonthlyBillingCycle } from "./utils";
+
+/**
+ * Calculates the total agent time usage in minutes for a specific team during their current billing period.
+ * The result is rounded up to 1 decimal places.
+ *
+ * @param teamDbId - The database ID of the team to calculate usage for
+ * @returns Promise that resolves to the total agent time usage in minutes
+ * @example
+ * const timeUsage = await calculateAgentTimeUsage(123);
+ * console.log(timeUsage); // 42.8
+ */
+export async function calculateAgentTimeUsage(teamDbId: number) {
+	const { start, end } = await getCurrentBillingPeriod(teamDbId);
+	const timeUsageMs = await agentTimeUsageMs(teamDbId, start, end);
+
+	return Math.ceil((timeUsageMs / 1000 / 60) * 10) / 10;
```

Comment:
IMO:

At first glance, since this involves time calculations that affect pricing, I thought it might be safer to use a library like the ones below.  
On the other hand, since it’s not directly calculating the amount, this approach might be fine as it is.

- [big\.js vs bignumber\.js vs decimal\.js vs mathjs vs numeraljs \| npm trends](https://npmtrends.com/big.js-vs-bignumber.js-vs-decimal.js-vs-mathjs-vs-numeraljs)

Reply:
I plan to use this method to calculate the pricing affecting value.
Therefore, I'll move the rounding logic closer to the UI layer.

Reply:
🛠️ [1a6d4d2](https://github.com/giselles-ai/giselle/pull/200/commits/1a6d4d2074fb0819d5ffd1f4c6c3181cc9238fc2)